core/src/main/java/org/apache/shiro/util/ThreadContext.java

```diff
@@ -327,26 +327,5 @@ public static void bind(Subject subject) {
     public static Subject unbindSubject() {
         return (Subject) remove(SUBJECT_KEY);
     }
-
-    private static final class InheritableThreadLocalMap<T extends Map<Object, Object>>
-                                            extends InheritableThreadLocal<Map<Object, Object>> {
-
-        /**
-         * This implementation was added to address a
-         * <a href="http://jsecurity.markmail.org/search/?q=#query:+page:1+mid:xqi2yxurwmrpqrvj+state:results">
-         * user-reported issue</a>.
```

Comment:
What issue was addressed here?

Reply:
This was a bug with the `InheriableThreadLocalMap` implementation, it needed to do a deep copy of the map. Since it's no longer Inheritable, the issue above no longer applies

Reply:
Great, thanks!